### DISCUSSION TITLE
Monotonic Constraint in Gradient Boosting

### DISCUSSION THREAD

**Discussion by SharmaLovely** [sebp/scikit-survival#496] [General]
Is there a way to add monotonic constraint in GradientBoostingSurvivalAnalysis?

**Comment by thierrymoudiki** [comment]
If I may, could you try this approach (still on a branch)? With isotonic regression as base learner (https://scikit-learn.org/stable/auto_examples/miscellaneous/plot_isotonic_regression.html) or any monotonic base learner: 

https://github.com/thierrymoudiki/scikit-survival/blob/master/examples/2024-11-14-scikit-survival.ipynb

**Comment by sebp** [comment]
We rely on scikit-learn's capabilities. Monotonic constraints are only supported in Histogram-based Gradient Boosting models. These are currently not supported by scikit-survival though (see 
https://github.com/sebp/scikit-survival/issues/473).
